lxd/images.go

```diff
@@ -2115,164 +2114,151 @@ func distributeImage(ctx context.Context, s *state.State, nodes []string, oldFin
 		return err
 	}
 
-	reverter := revert.New()
-	defer reverter.Fail()
 	for _, nodeAddress := range nodes {
 		if nodeAddress == localClusterAddress {
 			continue
 		}
 
 		var nodeInfo db.NodeInfo
 		err = s.DB.Cluster.Transaction(context.TODO(), func(ctx context.Context, tx *db.ClusterTx) error {
-			var err error
 			nodeInfo, err = tx.GetNodeByAddress(ctx, nodeAddress)
 			return err
 		})
 		if err != nil {
 			return fmt.Errorf("Failed to retrieve information about cluster member with address %q: %w", nodeAddress, err)
 		}
 
-		client, err := cluster.Connect(nodeAddress, s.Endpoints.NetworkCert(), s.ServerCert(), nil, true)
-		if err != nil {
-			return fmt.Errorf("Failed to connect to %q for image synchronization: %w", nodeAddress, err)
-		}
+		err = func() error {
+			client, err := cluster.Connect(nodeAddress, s.Endpoints.NetworkCert(), s.ServerCert(), nil, true)
+			if err != nil {
+				return fmt.Errorf("Failed to connect to %q for image synchronization: %w", nodeAddress, err)
+			}
 
-		client = client.UseTarget(nodeInfo.Name)
+			client = client.UseTarget(nodeInfo.Name)
 
-		resp, _, err := client.GetServer()
-		if err != nil {
-			logger.Error("Failed to retrieve information about cluster member", logger.Ctx{"err": err, "remote": nodeAddress})
-		} else {
-			vol := ""
-
-			val := resp.Config["storage.images_volume"]
-			if val != nil {
-				var ok bool
-				vol, ok = val.(string)
-				if !ok {
-					return errors.New("Invalid type for field \"storage.images_volume\"")
+			resp, _, err := client.GetServer()
+			if err != nil {
+				logger.Error("Failed to retrieve information about cluster member", logger.Ctx{"err": err, "remote": nodeAddress})
+			} else {
+				vol := ""
+
+				val := resp.Config["storage.images_volume"]
+				if val != nil {
+					var ok bool
+					vol, ok = val.(string)
+					if !ok {
+						return errors.New("Invalid type for field \"storage.images_volume\"")
+					}
 				}
-			}
 
-			skipDistribution := false
-
-			// If storage.images_volume is set on the cluster member, check if
-			// the image has already been downloaded to this volume. If so,
-			// skip distributing the image to this cluster member.
-			// If the option is unset, distribute the image.
-			if vol != "" {
-				for _, imageVolume := range imageVolumes {
-					if imageVolume == vol {
-						skipDistribution = true
-						break
+				skipDistribution := false
+
+				// If storage.images_volume is set on the cluster member, check if
+				// the image has already been downloaded to this volume. If so,
+				// skip distributing the image to this cluster member.
+				// If the option is unset, distribute the image.
+				if vol != "" {
+					for _, imageVolume := range imageVolumes {
+						if imageVolume == vol {
+							skipDistribution = true
+							break
```

Comment:
Would it make sense to not have that `skipDistribution` variable and simply `return nil` instead of `break`'ing and returning after the loop?

Reply:
fixed